core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -154,15 +155,17 @@ private[spark] class HighlyCompressedMapStatus private (
 
   override def writeExternal(out: ObjectOutput): Unit = Utils.tryOrIOException {
     loc.writeExternal(out)
-    emptyBlocks.writeExternal(out)
+    markedBlocks.writeExternal(out)
```

Comment:
I'd recommend investigating whether adding ``markedBlocks.runOptimize()`` right before ``markedBlocks.writeExternal`` reduces the size of the serialized output. Make sure to use version 0.5 or better of RoaringBitmap as well.